maven-model-builder/src/test/java/org/apache/maven/model/profile/DefaultProfileSelectorTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.model.profile;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.maven.api.model.Activation;
+import org.apache.maven.api.model.ActivationProperty;
+import org.apache.maven.api.model.Model;
+import org.apache.maven.api.model.Profile;
+import org.apache.maven.model.interpolation.ModelInterpolator;
+import org.apache.maven.model.profile.activation.PropertyProfileActivator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DefaultProfileSelectorTest {
+    private DefaultProfileSelector selector;
+    private ModelInterpolator interpolator;
+
+    @BeforeEach
+    public void setup() {
+        interpolator = Mockito.mock(ModelInterpolator.class);
+        selector = new DefaultProfileSelector(Collections.singletonList(new PropertyProfileActivator()), interpolator);
+    }
+
+    @Test
+    public void testProfileActivationInterpolation() {
```

Comment:
This test seems to work if the mock interpolator simply returns the input model, thereby indicating that the test is wrong or that the feature is already implemented.

Reply:
You're right, in that the test is only setting `property.name` on the activation, thus the property simply being set would cause the profile to be selected. I will address this first.